packages/flutter_tools/templates/create/android.tmpl/buildSrc/src/main/groovy/FlutterPlugin.groovy

```diff
@@ -0,0 +1,229 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.gradle
+
+import java.nio.file.Path
+import java.nio.file.Paths
+
+import com.android.builder.model.AndroidProject
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.Plugin
+import org.gradle.api.Task
+import org.gradle.api.file.CopySpec
+import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.bundling.Jar
+
+class FlutterPlugin implements Plugin<Project> {
```

Comment:
What's the plan to resolve https://github.com/flutter/flutter/issues/7504 ?  We might want to fix that issue before we make this template the default because if we need to change anything in this file, we'll be stuck without a fix for https://github.com/flutter/flutter/issues/7504